src/Symfony/Bundle/FrameworkBundle/CacheWarmer/AnnotationsCacheWarmer.php

```diff
@@ -28,17 +28,20 @@ class AnnotationsCacheWarmer extends AbstractPhpFileCacheWarmer
 {
     private $annotationReader;
     private $excludeRegexp;
+    private $debug;
 
     /**
      * @param Reader                 $annotationReader
      * @param string                 $phpArrayFile     The PHP file where annotations are cached
      * @param CacheItemPoolInterface $fallbackPool     The pool where runtime-discovered annotations are cached
+     * @param bool                   $debug            Run in debug mode
      */
-    public function __construct(Reader $annotationReader, $phpArrayFile, CacheItemPoolInterface $fallbackPool, $excludeRegexp = null)
+    public function __construct(Reader $annotationReader, $phpArrayFile, CacheItemPoolInterface $fallbackPool, $excludeRegexp = null, $debug)
```

Comment:
You've added a non-optional argument, which _should_ have broken tests. If there are none, can you add one to prevent future regressions? As this argument is non-optional, it would also be a BC break. Can you make this argument optional with a default value to mimic the previous behavior?

Reply:
All done.
Added test for cache warmer for both cases - debug and no debug
Made argument optional